### ISSUE TITLE
In Local mode, Batch Transform ManifestFile option gives the endpoint the ManifestFile as input, which seems incorrect

### ISSUE THREAD

**Issue by zmjjmz** [aws/sagemaker-python-sdk#826]
### System Information
- **Framework (e.g. TensorFlow) / Algorithm (e.g. KMeans)**: SKLearn, but I don't think that's important
- **Framework Version**: n/a
- **Python Version**: 2.7
- **CPU or GPU**: cpu
- **Python SDK Version**: 1.25.0
- **Are you using a custom image**: nope

### Describe the problem
Hey there, I'm developing a batch transform job and I'm trying to use the ManifestFile option as documented [here](https://sagemaker.readthedocs.io/en/stable/transformer.html#sagemaker.transformer.Transformer.transform). I'm assuming that the format of the ManifestFile is just a newline delimited text file, so it currently looks something like:

```
[{'prefix': 's3://okc-ml/'}, u'prototyping_data/svd_ubers_test/ds=2019-05-01/proc00_00000.parquet', u'prototyping_data/svd_ubers_test/ds=2019-05-01/proc00_00001.parquet', u'prototyping_data/svd_ubers_test/ds=2019-05-02/proc00_00002.parquet', u'prototyping_data/svd_ubers_test/ds=2019-05-02/proc00_00003.parquet']
```

The endpoint was reporting that the parquet files it was expecting were corrupted, so I had it print out what it received when it got that error, and lo and behold:

```
algo-1-7v3r8_1_2077bdfa2954 | Invalid Parquet file, got: b'\x80\x02T4\x01\x00\x00[{"prefix": "s3://okc-ml/"}, "prototyping_data/svd_ubers_test/ds=2019-05-01/proc00_00000.parquet", "prototyping_data/svd_ubers_test/ds=2019-05-01/proc00_00001.parquet", "prototyping_data/svd_ubers_test/ds=2019-05-02/proc00_00002.parquet", "prototyping_data/svd_ubers_test/ds=2019-05-02/proc00_00003.parquet"]q\x01.'
```

This shows, pretty clearly, that the endpoint is receiving the manifest file itself, rather than the URIs that it points to... Additionally, the batch transform outputs a file containing the 500 error response page at `output_path/{manifest_uri}.out`, which really further indicates to me that the manifest is being fed to the endpoint.

The documentation states that 

>’ManifestFile’ - the S3 URI points to a single manifest file listing each S3 object to use as
>    an input for the transform job.

which I took to mean that the Batch Transform agent would then retrieve those files and send them off to the endpoint. I'll add that that documentation should really point to the format of the Manifest File, as it took a bit of digging
### Minimal repro / logs
Please provide any logs and a bare minimum reproducible test case, as this will be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

- **Exact command to reproduce**:

I don't feel like putting the entrypoint up, but it shouldn't really matter, so here's a rough sketch of the API calls I'm making:
```
svd_estimator = SKLearn(                                                        
    entry_point='svd_entrypoint.py',                                            
    source_dir='sagemaker_source_dir',                                          
    train_instance_type='ml.c4.2xlarge',                                        
    role=sagemaker_session.get_caller_identity_arn(),                           
    sagemaker_session=sagemaker_session,
)                                                                               

svd_estimator.fit({'train': TRAINING_S3_URI})                  

TRANSFORMED_USER_PATH = 's3://okc-ml/prototyping_data/categoricals_test/try3/model='
transformer = svd_estimator.transformer(
    instance_count=1,
    instance_type='ml.c4.2xlarge',
    output_path=TRANSFORMED_USER_PATH,
    accept='application/x-parquet',
    max_concurrent_transforms=1,
    max_payload=100,
)
transformer.transform(
    data=MANIFEST_URI,
    data_type='ManifestFile',
    content_type='application/x-parquet',
)
```

That last call is where I encounter this issue.

**Comment by zmjjmz** [comment]
UPDATE: I'm amending the issue because some further testing showed that this only happens when the `sagemaker_session` is a `LocalSession`, i.e.

```
sagemaker_session = sagemaker.LocalSession(                                          
    boto_session=boto3.Session(profile_name=PROFILE))                      
```

When it's just a `sagemaker.Session` it works as expected.

**Comment by laurenyu** [comment]
Hello, thanks for your patience! A [quick search for "manifest"](https://github.com/aws/sagemaker-python-sdk/search?q=manifest+path%3Asrc&unscoped_q=manifest+path%3Asrc) turns up no results for Local Mode, so this is indeed a gap between Local Mode's Batch Transform support vs. using Batch Transform in SageMaker.

**Comment by ShwetaSingh801** [comment]
Hi @zmjjmz,

Thanks for using SageMaker and taking the time to suggest ways to improve SageMaker Python SDK. We have added your feature request to our backlog of feature requests and may consider putting it into future SDK versions. I will go ahead and close the issue now, please let me know if you have any more feedback. Let me know if you have any other questions.

Best,
Shweta

**Comment by laurenyu** [comment]
@ShwetaSingh801, @akrishna1995, and anyone else that has been recently closing these issues in favor of a private backlog: the decision back in the day to leave these issues open on GitHub was part of the attempt to embrace the open source nature of the SageMaker Python SDK. These public issues can let people make their voices heard on feature requests or even encourage contributors who want to help out but don't know where to start (hence the "contributions welcome" label). Just leaving my two cents as an outside observer – I don't have a stake in this, as it's been years since I worked in SageMaker and I don't use it myself.
